ui/pages/home/home.container.js

```diff
@@ -134,6 +139,29 @@ const mapStateToProps = (state) => {
   const shouldShowSeedPhraseReminder =
     selectedAccount && getShouldShowSeedPhraseReminder(state, selectedAccount);
 
+  const extensionCurrentVersion = semver.valid(process.env.METAMASK_VERSION);
+  const extensionMinimumVersion = semver.valid(
+    remoteFeatureFlags.extensionMinimumVersion,
+  );
+  const isExtensionOutdated =
+    extensionCurrentVersion && extensionMinimumVersion
+      ? semver.lt(extensionCurrentVersion, extensionMinimumVersion)
+      : false;
+
+  const updateModalCooldown = 24 * 60 * 60 * 1000; // 24 hours
+  const enoughTimePassedSinceLastDismissal = updateModalLastDismissedAt
+    ? Date.now() - updateModalLastDismissedAt > updateModalCooldown
+    : true;
+  const enoughTimePassedSinceLastUpdate = lastUpdatedAt
+    ? Date.now() - lastUpdatedAt > updateModalCooldown
+    : true;
+
+  const showUpdateModal =
+    isExtensionOutdated &&
+    isUpdateAvailable &&
+    enoughTimePassedSinceLastDismissal &&
+    enoughTimePassedSinceLastUpdate;
+
```

Comment:
nite: it'd be nice if this logic were moved out into its own helper function like `shouldShowSeedPhraseReminder` and `showWhatsNewPopup` do.